lib/wasi/Cargo.toml

```diff
@@ -10,22 +10,52 @@ license = "MIT"
 readme = "README.md"
 edition = "2018"
 
+[lib]
+crate-type = ["cdylib", "rlib"]
+
 [dependencies]
-bincode = "1"
+cfg-if = "1.0"
 thiserror = "1"
 generational-arena = { version = "0.2", features = ["serde"] }
-libc = { version = "^0.2", default-features = false }
 tracing = { version = "0.1" }
-getrandom = "0.2"
-typetag = "0.1"
-serde = { version = "1.0", features = ["derive"] }
+getrandom = { version = "0.2" }
+typetag = { version = "0.1", optional = true }
+serde = { version = "1.0", default-features = false, features = ["derive"], optional = true }
 wasmer-wasi-types = { path = "../wasi-types", version = "2.0.0" }
-wasmer = { path = "../api", version = "2.0.0", default-features = false, features = ["sys"] }
+wasmer = { path = "../api", version = "2.0.0", default-features = false }
+wasmer-vfs = { path = "../vfs", version = "2.0.0", default-features = false }
 
 [target.'cfg(windows)'.dependencies]
 winapi = "0.3"
 
+[target.'cfg(unix)'.dependencies]
+libc = { version = "^0.2", default-features = false }
+
 [features]
-default = ["logging"]
+default = ["logging", "host_fs"]
+enable-serde = [
+    "typetag",
+    "serde",
+]
 logging = ["tracing/log"]
-disable-all-logging = ["tracing/release_max_level_off", "tracing/max_level_off"]
\ No newline at end of file
+disable-all-logging = [
+    "tracing/release_max_level_off",
+    "tracing/max_level_off"
+]
+host_fs = ["wasmer-vfs/host_fs"]
+mem_fs = ["wasmer-vfs/mem_fs"]
+js = [
```

Comment:
Just like we did with `wasmer` (`lib/api/`), I think it's good to differentiate between `sys` and `js`. Thought?